### PR TITLE
JW-928 Tests for Agent service

### PR DESCRIPTION
**Pull request by katrybacka**

### CODE REVIEW

**Comment by katrybacka** [comment]
Janie, przetestuj proszę

File: CIScripts/Tests/AgentServiceTests.ps1 [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+    function Disable-AgentService {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session)
+
+        Invoke-Command -Session $Session -ScriptBlock {
+            Stop-Service ContrailAgent | Out-Null
+        }
+    }
+
+    function Test-isAgentServiceRegistered {
```
Comment (m-kostrzewa) [inline CIScripts/Tests/AgentServiceTests.ps1]: `is` should be `Is` ;)

Comment (m-kostrzewa) [reply CIScripts/Tests/AgentServiceTests.ps1]: just a reminder

File: CIScripts/Tests/AgentServiceTests.ps1 [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    function Test-AgentServiceIsUnregisteredAfterUninstall {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        Write-Host "===> Running: Test-AgentServiceIsUnregisteredAfterUninstall"
+
+        Write-Host "======> Given Agent is installed"
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Install-Agent -Session $Session
```
Comment (jacek-izykowski) [inline CIScripts/Tests/AgentServiceTests.ps1:172]: Do we need to install agent here?

File: CIScripts/Tests/AgentServiceTests.ps1 [hunk 3]
```diff
@@ -0,0 +1,255 @@
+function Test-AgentService {
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+    . $PSScriptRoot\CommonTestCode.ps1
+    . $PSScriptRoot\..\Job.ps1
+
+    $AgentServiceTestsTimeTracker = [Job]::new("Test-AgentService")
```
Comment (jablonskim) [inline CIScripts/Tests/AgentServiceTests.ps1:8]: I think that we should not create new TimeTracker here. We should use existing Job like in this commit https://github.com/codilime/contrail-windows-tools/commit/962af2d37b14dc4a94448d381e279ffc496119f3

File: CIScripts/Tests/AgentServiceTests.ps1 [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    function Test-AgentServiceDisabling {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        Write-Host "===> Running: Test-AgentServiceDisabling"
+
+        Write-Host "======> Given Agent is installed and enabled"
+        Clear-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+        Install-Agent -Session $Session
```
Comment (jacek-izykowski) [inline CIScripts/Tests/AgentServiceTests.ps1:218]: Do we need to install agent here?

File: CIScripts/Tests/AgentServiceTests.ps1 [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+    function Test-IsAgentServiceRegistered {
+        Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session)
+
+        $Service = Invoke-Command -Session $Session -ScriptBlock {
+            return $(if (Get-Service "ContrailAgent" -ErrorAction SilentlyContinue) { $true } else { $false }) 
+        }
+        return $Service        
+    }
+    function Assert-IsAgentServiceRegistered {
```
Comment (jablonskim) [inline CIScripts/Tests/AgentServiceTests.ps1]: Please add new empty line here

**Review by jablonskim — CHANGES_REQUESTED** [review]

File: CIScripts/Tests/AgentServiceTests.ps1 (hunk 2 above)
Comment (katrybacka) [reply CIScripts/Tests/AgentServiceTests.ps1:172]: If we don't want to rely on sequence of tests and we want to use them in different order someday we have to make sure that Agent is installed. In my opinion this test is constructed to be able to run it independently.

File: CIScripts/Tests/AgentServiceTests.ps1 (hunk 4 above)
Comment (katrybacka) [reply CIScripts/Tests/AgentServiceTests.ps1:218]: Same here

File: CIScripts/Tests/AgentServiceTests.ps1 [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+        Test-AgentServiceIsUnregisteredAfterUninstall -Session $Session -TestConfiguration $TestConfiguration
+    })
+    $AgentServiceTestsTimeTracker.StepQuiet("Test-AgentServiceEnabling", {
+        Test-AgentServiceEnabling -Session $Session -TestConfiguration $TestConfiguration
+    })
+    $AgentServiceTestsTimeTracker.StepQuiet("Test-AgentServiceDisabling", {
+        Test-AgentServiceDisabling -Session $Session -TestConfiguration $TestConfiguration
+    })
+
+    # Test cleanup
```
Comment (jacek-izykowski) [inline CIScripts/Tests/AgentServiceTests.ps1:254]: If you do not wish to assume any specific order of invoking tests (as you mentioned earlier), please make sure that Agent is installed here (some tests leave it uninstalled).

**Review by jacek-izykowski — APPROVED** [review]

**Review by jablonskim — APPROVED** [review]

**Pull request merged**